src/app/account/mfa/sms/page.tsx

```diff
@@ -1,199 +1,186 @@
-                                    <Button onClick={() => phone.destroy()}>Remove from account</Button>
-                                </div>
-                            )}
-                        </li>
-                    )
-                })}
-            </ul>
-        </>
-    )
-}
-
-// Generate and display backup codes
-function GenerateBackupCodes() {
-    const { user } = useUser()
-    const [backupCodes, setBackupCodes] = React.useState<BackupCodeResource | undefined>(undefined)
-
-    const [loading, setLoading] = React.useState(false)
-
-    React.useEffect(() => {
-        if (backupCodes) {
+    async function handleVerify() {
+        if (!user || !phoneResourceToVerify || !verificationCode) {
+            setError('Verification code is required.')
             return
         }
 
-        setLoading(true)
-        void user
-            ?.createBackupCode()
-            .then((backupCode: BackupCodeResource) => {
-                setBackupCodes(backupCode)
-                setLoading(false)
-            })
-            .catch((err) => {
-                // See https://clerk.com/docs/custom-flows/error-handling
-                // for more info on error handling
-                console.error(JSON.stringify(err, null, 2))
-                setLoading(false)
-            })
-    }, []) // eslint-disable-line react-hooks/exhaustive-deps
-
-    if (loading) {
-        return <p>Loading...</p>
-    }
-
-    if (!backupCodes) {
-        return <p>There was a problem generating backup codes</p>
+        setIsVerifying(true)
+        setError('')
+
+        try {
+            const verifiedPhoneResource = await phoneResourceToVerify.attemptVerification({ code: verificationCode })
+
+            if (verifiedPhoneResource.verification.status === 'verified') {
+                await verifiedPhoneResource.setReservedForSecondFactor({ reserved: true })
+                await verifiedPhoneResource.makeDefaultSecondFactor()
+
+                setVerificationSuccess(true)
+                await user.reload()
+            } else {
+                setError('Verification failed. Please try again.')
+            }
+        } catch (err) {
+            // err is implicitly 'unknown'
+            logger.error({ err, message: 'Error verifying code' })
+            // Set a generic error message for the user
+            setError('Invalid verification code. Please try again.')
+            setVerificationSuccess(false)
+        } finally {
+            setIsVerifying(false)
+        }
     }
 
-    return (
-        <ol>
-            {backupCodes.codes.map((code, index) => (
-                <li key={index}>{code}</li>
-            ))}
-        </ol>
-    )
-}
-
-export default function ManageSMSMFA() {
-    const [showBackupCodes, setShowBackupCodes] = React.useState(false)
-    const { openUserProfile } = useClerk()
-    const { isLoaded, user } = useUser()
-
     if (!isLoaded) return null
 
     if (!user) {
```

Comment:
I think this can be brought up before all the other function definitions, right after your `useState` calls. we should probably be redirecting too since they can't access this page at all if there's no user?

Moving this up would also allow all your other `user?` references to be just `user` as there's no null possibility. I would also move the `if (!isLoaded) return null` up too